azurerm/resource_arm_loadbalancer_probe.go

```diff
@@ -189,19 +201,24 @@ func resourceArmLoadBalancerProbeRead(d *schema.ResourceData, meta interface{})
 
 	d.Set("name", config.Name)
 	d.Set("resource_group_name", id.ResourceGroup)
-	d.Set("protocol", config.ProbePropertiesFormat.Protocol)
-	d.Set("interval_in_seconds", config.ProbePropertiesFormat.IntervalInSeconds)
-	d.Set("number_of_probes", config.ProbePropertiesFormat.NumberOfProbes)
-	d.Set("port", config.ProbePropertiesFormat.Port)
-	d.Set("request_path", config.ProbePropertiesFormat.RequestPath)
-
-	var load_balancer_rules []string
-	if config.ProbePropertiesFormat.LoadBalancingRules != nil {
-		for _, ruleConfig := range *config.ProbePropertiesFormat.LoadBalancingRules {
-			load_balancer_rules = append(load_balancer_rules, *ruleConfig.ID)
+
+	if properties := config.ProbePropertiesFormat; properties != nil {
+		d.Set("protocol", properties.Protocol)
+		d.Set("interval_in_seconds", properties.IntervalInSeconds)
+		d.Set("number_of_probes", properties.NumberOfProbes)
+		d.Set("port", properties.Port)
+		d.Set("request_path", properties.RequestPath)
+
+		var load_balancer_rules []string
+		if rules := properties.LoadBalancingRules; rules != nil {
+			for _, ruleConfig := range *rules {
+				if id := ruleConfig.ID; id != nil {
+					load_balancer_rules = append(load_balancer_rules, *id)
+				}
+			}
 		}
+		d.Set("load_balancer_rules", load_balancer_rules)
```

Comment:
Nitpick: This is a trivial case (currently) but best practice when using `d.Set()` with aggregate types (TypeList, TypeSet, TypeMap), [we should perform error checking to prevent issues where the code is not properly able to set the Terraform state](https://www.terraform.io/docs/extend/best-practices/detecting-drift.html#error-checking-aggregate-types). e.g.

```go
if err := d.Set("load_balancer_rules", load_balancer_rules); err != nil {
  return fmt.Errorf("error setting load_balancer_rules: %s", err)
}
```